### PR TITLE
Deployment tooling update - bin/eb-task-run

### PR DESCRIPTION
**Pull request by indigobravo**
The bin/eb-task-run script has been updated to handle multi-region
deployments.

### CODE REVIEW

File: bin/eb-task-run [hunk 1]
```diff
@@ -27,7 +27,7 @@ status () {
     echo "--->" "$@" >&2
 }
 
-if [ "$#" -ne 4 ]; then
+if [ "$#" -ne 5 ]; then
```
Comment (indigobravo) [inline bin/eb-task-run:30]: Increased mandatory parameters to 5 now we are passing in `$REGION`

File: bin/eb-task-run [hunk 2]
```diff
@@ -38,9 +38,10 @@ APP=$1
 ENV=$2
 TIMEOUT=$3
 COMMAND=$4
+REGION=$5
```
Comment (indigobravo) [inline bin/eb-task-run:41]: Set `REGION` to $5

File: bin/eb-task-run [hunk 3]
```diff
@@ -38,9 +38,10 @@ APP=$1
...
 
 
-if ! eb-env-exists "$APP" "$ENV"; then
+if ! eb-env-exists "$APP" "$ENV" "$REGION"; then
```
Comment (indigobravo) [inline bin/eb-task-run:44]: Added `$REGION` to `eb-env-exists` call

File: bin/eb-task-run [hunk 4]
```diff
@@ -53,7 +54,8 @@ INSTANCE_ID=$(aws ec2 describe-instances \
                   --filters "Name=tag:elasticbeanstalk:environment-name,Values=${APP}-${ENV}" \
                             "Name=instance-state-name,Values=running" \
                   --query "Reservations[0].Instances[0].InstanceId" \
-                  --output text)
+                  --output text \
+	          --region "$REGION")
```
Comment (indigobravo) [inline bin/eb-task-run:58]: Added `--region` parameter to `aws` command.

File: bin/eb-task-run [hunk 5]
```diff
@@ -74,18 +76,21 @@ remove_instance_protection() {
   echo aws autoscaling set-instance-protection \
       --auto-scaling-group-name "$asg_name" \
       --instance-ids "$instance_id" \
-      --no-protected-from-scale-in
+      --no-protected-from-scale-in \
+      --region "$REGION"
```
Comment (indigobravo) [inline bin/eb-task-run:80]: Added `--region` parameter to `aws` command

File: bin/eb-task-run [hunk 6]
```diff
@@ -74,18 +76,21 @@ remove_instance_protection() {
...
 }
 
 ASG_NAME=$(aws elasticbeanstalk describe-environment-resources \
                --environment-name "${APP}-${ENV}" \
                --query "EnvironmentResources.AutoScalingGroups[0].Name" \
-               --output text)
+               --output text \
+               --region "$REGION")
```
Comment (indigobravo) [inline bin/eb-task-run:87]: Added `--region` parameter to `aws` command

File: bin/eb-task-run [hunk 7]
```diff
@@ -74,18 +76,21 @@ remove_instance_protection() {
...
 
 aws autoscaling set-instance-protection \
     --auto-scaling-group-name "$ASG_NAME" \
     --instance-ids "$INSTANCE_ID" \
-    --protected-from-scale-in
+    --protected-from-scale-in \
+    --region "$REGION"
```
Comment (indigobravo) [inline bin/eb-task-run:93]: Added `--region` parameter to `aws` command

File: bin/eb-task-run [hunk 8]
```diff
@@ -102,7 +107,8 @@ COMMAND_ID=$(aws ssm send-command \
                  }" \
                  --query 'Command.CommandId' \
                  --output-s3-bucket-name "$OUTPUT_BUCKET_NAME" \
-                 --output text)
+                 --output text \
+	         --region "$REGION")
```
Comment (indigobravo) [inline bin/eb-task-run:111]: Added `--region` parameter to `aws` command

File: bin/eb-task-run [hunk 9]
```diff
@@ -102,7 +108,8 @@ COMMAND_ID=$(aws ssm send-command \
...
 
 status "running command $COMMAND_ID. Output is available in the \"$OUTPUT_BUCKET_NAME\" S3 bucket."
-eb-task-wait "$COMMAND_ID" "$INSTANCE_ID"
+eb-task-wait "$COMMAND_ID" "$INSTANCE_ID" "$REGION"
```
Comment (indigobravo) [inline bin/eb-task-run:115]: Pass `$REGION` to `eb-task-wait`

**Review by robertknight — APPROVED** [review]

**Pull request merged**